app/Models/Chat/UserChannel.php

```diff
@@ -23,6 +23,14 @@ class UserChannel extends Model
 {
     protected $primaryKeys = ['user_id', 'channel_id'];
 
+    private ?int $lastReadIdToSet;
+
+    public function getLastReadIdAttribute($value): ?int
+    {
+        // return the value we tried to set it to, not the builder.
+        return $this->lastReadIdToSet ?? $value;
```

Comment:
only return `lastReadIdToSet` if `$value` is a ~~builder~~ query expression instead?